File: 01_StudentCode/01_A/3A/computer_science_exercises/02_controlFlow/controlFlow.py
# Control Flow, ryan Kelley, v0.1 

# DECLARATIONS 
favColor = "Blue"
luckyNumber = 25
myGPA = 3.0
myAge = 66 
pineappleOnPizza = True 

# if Statement 
if luckyNumber > 30:
    print("Wow, what a great number!")

if favColor != "Green":
    print("Your favorite color stinks.")

# if-else Statement  
if myGPA >= 3.0: 
    print("Congratulations on making the honor roll!")
else: 
    print("You did not make the honor roll.")

# if-elif-else Statement 
# if myAge > 65: 
#     print("Please enjoy retirement!")
# elif myAge > 55: 
#     print("You have earned a $10,000 bonus!")
# elif myAge > 40: 
#     print("You have earned a $5,000 bonus.")
# else:
#     print("You have earned a $1,000 bonus.")
# 99% OF THE TIME CHECK FOR HIGHEST VALUES FIRST! 

if myAge > 40: 
    print("You have earned a $5,000 bonus.")
elif myAge > 55: 
    print("You have earned a $10,000 bonus!")
elif myAge > 65: 
    print("Please enjoy retirement!")    
else:
    print("You have earned a $1,000 bonus.")

# NESTED IF / ELIF / ELSE STATEMENTS    
if pineappleOnPizza == True: 
    print("You have strange tastes.  I bet you're old.")
    if myAge > 50:
        print("What was it like riding to school on a dinosaur?")                
    else: 
        print("Ok, you're still young enough to be cool.")
else:
    print("Ok good, you eat pizza like a normal human.")    
     

File: 01_StudentCode/01_A/4A/computer_science_exercises/02_controlFlow/controlFlow.cs
// Ryan Kelley, Program Template, v0.01
using System; 

namespace controlFlow 
{
    class controlFlow 
    {
        static void Main(string[] args)
        {
            // DECLARATIONS 
            string favColor = "Blue"; 
            int luckyNumber = -5; 
            float myGPA = 1.33F;
            int myAge = 43; 
            bool pineappleOnPizza = true; 

            // if Statements -- Check for a single condition!
            if (favColor == "Green") {
                Console.WriteLine("Green with envy!");
            }

            if (luckyNumber % 2 == 0 ) {
                Console.WriteLine("Your lucky number is even!");
            }

            // if - else Statement -- Checks for a single condition, but has two actions. 
            if (myAge > 15) {
                Console.WriteLine("You are eligible to drive");                
            } else {
                Console.WriteLine("START WALKING SCRUB!");
            }

            // if -- else if -- else -- Checks multiple outcomes. 
            if (myGPA == 4.0F) {
                Console.WriteLine("Congrats on straight A grades!");                
            } else if (myGPA >= 3.0F) {
                Console.WriteLine("Congrats on making honor roll!");                
            } else if (myGPA >= 2.0F) {
                Console.WriteLine("You are graduation ready!");                
            } else {
                Console.WriteLine("You should probably study!");                
            }
            // WHEN CHECKING NUMBER VALUES, START WITH THE HIGHEST VALUE!!!!

            if (myGPA >= 2.0F) {
                Console.WriteLine("You are graduation ready!");                     
            } else if (myGPA >= 3.0F) {
                Console.WriteLine("Congrats on making honor roll!");                
            } else if (myGPA == 4.0F) {
                Console.WriteLine("Congrats on straight A grades!");                           
            } else {
                Console.WriteLine("You should probably study!");                
            }

            // Nested Statements 
            if (pineappleOnPizza == true) {
                Console.WriteLine("Eww, that's gross. You must be a boomer.  How old are you?");
                if (myAge > 60) {
                    Console.WriteLine("Just as I suspected! What was it like having a dinosaur growing up?");
                } else {
                    Console.WriteLine("Ok, so you're not a boomer but still have no taste in food.");
                }
            } else {
                Console.WriteLine("Glad to see you have common sense!"); 
            }

        }
    }
}
